Bug: Getting error during IOM deployment

### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

5.2

### Module Version

2.11.0.1023

### PowerShell Version

7.3.6

### PowerCLI Version

13.1.0.21624340

### PowerVCF Version

2.4.1.1000

### Guest Operating System

windows server 2019

### Environment Details

_No response_

### Description

Issue while running Invoke-IOMdeployment at step Activating the VMware Cloud Foundation Integration in VMware Aria Operations

### Error or Debug Output

'''Powershell
Error at Script Line 19290
 [06-20-2024_05:58:43] [INFO]  Relevant Command: $adapterDetails = Invoke-RestMethod -Method GET -Uri $uri -Headers $vropsBasicHeaders
 [06-20-2024_05:58:43] [INFO]  Error Message: Cannot convert 'System.Object[]' to the type 'System.Uri' required by parameter 'Uri'. Specified method is not supported.
 [06-20-2024_05:58:43] [ERROR] Cannot convert 'System.Object[]' to the type 'System.Uri' required by parameter 'Uri'. Specified method is not supported.
 [06-20-2024_05:58:43] [ERROR] Cannot convert 'System.Object[]' to the type 'System.Uri' required by parameter 'Uri'. Specified method is not supported.
 [06-20-2024_05:58:43] [ERROR] Cannot convert 'System.Object[]' to the type 'System.Uri' required by parameter 'Uri'. Specified method is not supported.
 [06-20-2024_05:58:43] [ERROR] Cannot convert 'System.Object[]' to the type 'System.Uri' required by parameter 'Uri'. Specified method is not supported.
 [06-20-2024_05:58:43] [ERROR] Cannot convert 'System.Object[]' to the type 'System.Uri' required by parameter 'Uri'. Specified method is not supported.
'''

### Expected Behavior

There should not any errors while running Invoke-IOMdeployment

### Actual Behavior

Error while  Activating the VMware Cloud Foundation Integration in VMware Aria Operations

### Steps to Reproduce

Run Invoke-IOMdeployment and it will fail at step Activating the VMware Cloud Foundation Integration in VMware Aria Operations

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -52,6 +52,7 @@
 - Enhanced `Import-vROPSNotification` cmdlet to consume a JSON file and pass in -alertPluginName and -emailAddress parameters.
 - Enhanced `Invoke-DriDeployment` cmdlet to support v1.26 for Tanzu distribution.
 - Enhanced `Export-GlobalWsaJsonSpec` cmdlet to include the export from Planning and Preparations Workbook the Workspace ONE Access database IP.
+- Enhanced `Register-vROPSManagementPack` cmdlet to handle the new VCFDiagnostics management pack in VMware Aria Operations.
 - Converted `aria-operations-notifications-vcf.csv` to 'aria-operations-notifications-vcf.json'.
 - Converted `aria-operations-notifications-srm.csv' to 'aria-operations-notifications-srm.json'.
 

--- a/PowerValidatedSolutions.psd1
+++ b/PowerValidatedSolutions.psd1
@@ -11,7 +11,7 @@
     RootModule = 'PowerValidatedSolutions.psm1'
 
     # Version number of this module.
-    ModuleVersion = '2.11.0.1026'
+    ModuleVersion = '2.11.0.1027'
     # ID used to uniquely identify this module
     GUID              = 'a6dfed7b-65d2-4da2-bdcc-7f3d3df9b75d'
 

--- a/PowerValidatedSolutions.psm1
+++ b/PowerValidatedSolutions.psm1
@@ -19281,19 +19281,20 @@ Function Register-vROPSManagementPack {
                 if (($vcfVropsDetails = Get-vROPsServerDetail -fqdn $server -username $user -password $pass)) {
                     if (Test-vROPSConnection -server $vcfVropsDetails.loadBalancerFqdn) {
                         if (Test-vROPSAuthentication -server $vcfVropsDetails.loadBalancerFqdn -user $vcfVropsDetails.adminUser -pass $vcfVropsDetails.adminPass) {
+                            if ($packType -eq "VCF") {$packTypeUrl = "$packType-"} else {$packTypeUrl = "$packType"}
                             # Connect to VMware Aria Operations and extract the Management Pack Details
                             $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $($vcfVropsDetails.adminUser), $($vcfVropsDetails.adminPass)))) # Create Basic Authentication Encoded Credentials
                             $vropsBasicHeaders = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
                             $vropsBasicHeaders.Add("Authorization", "Basic $base64AuthInfo")
                             $vropsBasicHeaders.Add("Content-Type", "application/json")
-                            $uri = ((Get-vROPSManagementPack -server $vcfVropsDetails.loadBalancerFqdn -username $vcfVropsDetails.adminUser -password $vcfVropsDetails.adminPass | Where-Object { $_.links -match $packType }).links | Where-Object { $_.rel -eq "pak_information" }).href
-                            $adapterDetails = Invoke-RestMethod -Method GET -Uri $uri -Headers $vropsBasicHeaders
-                            $uri = ((Get-vROPSManagementPack -server $vcfVropsDetails.loadBalancerFqdn -username $vcfVropsDetails.adminUser -password $vcfVropsDetails.adminPass | Where-Object { $_.links -match $packType }).links | Where-Object { $_.rel -eq "pak_cluster_status" }).href
+                            $uri = ((Get-vROPSManagementPack -server $vcfVropsDetails.loadBalancerFqdn -username $vcfVropsDetails.adminUser -password $vcfVropsDetails.adminPass | Where-Object { $_.links -match $packTypeUrl }).links | Where-Object { $_.rel -eq "pak_information" }).href
+                            $adapterDetails = Invoke-RestMethod -Method GET -Uri $uri -Headers $vropsBasicHeaders -SkipCertificateCheck -UseBasicParsing
+                            $uri = ((Get-vROPSManagementPack -server $vcfVropsDetails.loadBalancerFqdn -username $vcfVropsDetails.adminUser -password $vcfVropsDetails.adminPass | Where-Object { $_.links -match $packTypeUrl }).links | Where-Object { $_.rel -eq "pak_cluster_status" }).href
                             if ($state -eq "enable") {
                                 if (!(Get-vROPSSolution | Where-Object { $_.id -match $packType })) {
                                     Set-vROPSManagementPack -server $vcfVropsDetails.loadBalancerFqdn -username $vcfVropsDetails.adminUser -password $vcfVropsDetails.adminPass -pakId ((($adapterDetails.pak_id) -Split ("-"))[0]) -version $adapterDetails.version -status enable | Out-Null
                                     Do {
-                                        $status = Invoke-RestMethod -Method GET -Uri $uri -Headers $vropsBasicHeaders
+                                        $status = Invoke-RestMethod -Method GET -Uri $uri -Headers $vropsBasicHeaders -SkipCertificateCheck -UseBasicParsing
                                     } Until ( $status.cluster_pak_install_status -ne "CANDIDATE" )
                                     if ($status.cluster_pak_install_status -eq "COMPLETED") {
                                         Write-Output "Activating ($packType) Management Pack on VMware Aria Operations ($($vcfVropsDetails.loadBalancerFqdn)): SUCCESSFUL"

